gossip/evmstore/store.go

```diff
@@ -119,7 +123,6 @@ func (s *Store) RebuildEvmSnapshot(root common.Hash) {
 	if s.Snaps == nil {
```

Comment:
Question: if we check like that then when the snapshot rebuild can be processed?

Reply:
If s.Snaps is nil, then it'll have to call GenerateEvmSnapshot with `rebuild`=true and `async`=true. It should have almost the same effect as calling s.Snaps.Rebuild

Reply:
Yes. I mean if the s.Snaps is nil then it never generate a new snapshot with new hash root?

Reply:
If the s.Snaps is not nil, then it'll call s.Snaps.Rebuild, which will erase existing data (if any) and generate new snapshot in background with new hash root. We'll proceed straight to full syncing, each new processed block will update generation of the snapshot. Eventually, snapshot should be generated during full syncing